conformance/lang/expressions/raw_template_expr.balt

```diff
@@ -0,0 +1,690 @@
+                        a}b`; // @error incompatible interpolation: expected 'int', found 'string'
+}
+
+
+
+Test-Case: error
+Description: Test incompatible types in raw template expression constructing an object belonging to the abstract RawTemplate object type, with a user-defined class 
+             of a type that is a subtype of `object:RawTemplate` as the contextually-expected type.
+Fail-Issue: ballerina-platform/ballerina-lang#36200
+Labels: array-type, intersection-type, list-constructor-expr, module-class-defn, object:RawTemplate, object-type, raw-template-expr
+
+class MyRawTemplate {
+    *object:RawTemplate;
+
+    public readonly & string[] strings = [];
+    public int[] insertions = [];
+
+    function init(string[] strings, int[] insertions) {
+        self.strings = strings.cloneReadOnly();
+        self.insertions = insertions;
+    }
+}
+
+function errorFunction() {
+    string a = "two";
+    MyRawTemplate b = `ab${ // @error incompatible value for `strings`: expected 'string:Char', found 'string'
+                       "1"} ${ // @error incompatible interpolation: expected 'int', found 'string'
+                        a}b`; // @error incompatible interpolation: expected 'int', found 'string'
+}
+
+Test-Case: output
+Description: Test the raw template `strings` and `insertions` fields being read-only when required by the contextually-expected type.
+Fail-Issue: ballerina-platform/ballerina-lang#36200
+Labels: array-type, intersection-type, list-constructor-expr, module-class-defn, object:RawTemplate, raw-template-expr
+
+class MyRawTemplate {
+    *object:RawTemplate;
+
+    public final readonly & string[] strings = [];
+    public final readonly & int[] insertions = [];
+}
+
+function init() {
+    int i = 1;
+    MyRawTemplate a = `number ${i} and number ${2}`;
+
+    io:println(a is object:RawTemplate); // @output true
+    io:println(a.strings); // @output ["number "," and number ",""]
+    io:println(a.insertions); // @output [1,2]
+
+    object:RawTemplate b = a;
+    io:println(b is MyRawTemplate); // @output true
+}
+
+Test-Case: panic
+Description: Test the raw template `strings` fields being read-only when required by the contextually-expected type.
+Fail-Issue: ballerina-platform/ballerina-lang#36200
+Labels: array-type, intersection-type, list-constructor-expr, module-class-defn, object:RawTemplate, raw-template-expr
+
+class MyRawTemplate {
+    *object:RawTemplate;
+
+    public final readonly & string[] strings = [];
+    public final readonly & int[] insertions = [];
+}
+
+function init() {
+    int i = 1;
+    MyRawTemplate a = `number ${i} and number ${2}`;
+    object:RawTemplate b = a;
+    b.strings = []; // @panic cannot overwrite the read-only `strings` field
+}
+
+Test-Case: panic
+Description: Test the raw template `insertions` fields being read-only when required by the contextually-expected type.
```

Comment:
Shall we also add a positive test for this where contextually-expected type doesn't require `readonly` and can update the `insertions` fields?